packages/client/src/v2-events/features/events/useEvents/procedures/actions/action.ts

```diff
@@ -212,6 +212,13 @@ export function useEventAction<P extends DecorateMutationProcedure<any>>(
     ...mutationOptions
   })
 
+  const notifyMutation = useMutation({
+    ...trpcOptionsProxy.event.actions.notify.mutationOptions(),
+    ...queryClient.getMutationDefaults(
+      trpcOptionsProxy.event.actions.notify.mutationKey()
+    )
+  })
```

Comment:
`useEventAction` is a generic wrapper for any type of action (including notify). Can you lift the 

```ts
if (actionType === ActionType.DECLARE && params.metadata.incomplete) {
```

logic all the way up up the chain to where `events.actions.declare.mutate` is triggered from?

If I'm not mistaken, your problem with having to take the `metadata.incomplete = true` information from `useReviewActionConfig.tsx` downwards will also then go away